### PR TITLE
use \copy for getTables

### PR DESCRIPTION
**Pull request by pivnicek**
hopefully fixes https://github.com/keboola/internal/issues/198

### CODE REVIEW

**Comment by pivnicek** [comment]
Going to get rid of default values too as they aren't used yet

File: src/Keboola/DbExtractor/Extractor/PgSQL.php [hunk 1]
```diff
@@ -415,44 +413,50 @@ function ($table) {
             );
         }
 
-        $res = $this->db->query($sql);
+        $this->runCopyCommand($sql, "tables.csv");
+
+        $tablesCsv = new CsvFile("tables.csv");
```
Comment (pivnicek) [inline src/Keboola/DbExtractor/Extractor/PgSQL.php]: Not sure about naming this file, maybe should use system temp dir?

File: src/Keboola/DbExtractor/Extractor/PgSQL.php [hunk 2]
```diff
@@ -376,17 +376,15 @@ public function getTables(?array $tables = null): array
...
-    LEFT JOIN pg_catalog.pg_attrdef d ON (a.attrelid, a.attnum) = (d.adrelid,  d.adnum) -- default values
+    JOIN pg_class c ON a.attrelid = c.oid AND c.reltype != 0 
+    INNER JOIN pg_namespace ns ON ns.oid = c.relnamespace 
+    LEFT JOIN pg_index i ON a.attrelid = i.indrelid AND a.attnum = ANY(i.indkey) 
     WHERE 
-      NOT a.attisdropped -- exclude dropped columns
-      AND a.attnum > 0 -- exclude system columns
-      AND ns.nspname != 'information_schema' -- exclude system namespaces
+      NOT a.attisdropped 
+      AND a.attnum > 0
```
Comment (odinuv) [inline src/Keboola/DbExtractor/Extractor/PgSQL.php]: the comments seemed useful

File: src/Keboola/DbExtractor/Extractor/PgSQL.php (hunk 1 above)
Comment (odinuv) [reply src/Keboola/DbExtractor/Extractor/PgSQL.php]: would be definitely better, this might create files in folder not available for writing

Comment (odinuv) [reply src/Keboola/DbExtractor/Extractor/PgSQL.php]: i think `sys_get_temp_dir() . '/' . uniqid('tables-csv')`  should be pretty safe

File: src/Keboola/DbExtractor/Extractor/PgSQL.php [hunk 3]
```diff
@@ -461,6 +465,39 @@ function ($table) {
...
+            "PGPASSWORD='%s' psql -h %s -p %s -U %s -d %s -w -c %s",
+            $this->dbConfig['password'],
+            $this->dbConfig['host'],
+            $this->dbConfig['port'],
+            $this->dbConfig['user'],
+            $this->dbConfig['database'],
+            escapeshellarg(
+                sprintf(
+                    $copyCommand,
+                    rtrim($query, '; '),
```
Comment (odinuv) [inline src/Keboola/DbExtractor/Extractor/PgSQL.php]: how about `rtrim(rtrim($query), ';')`  if the whitespace is a problem?

File: src/Keboola/DbExtractor/Extractor/PgSQL.php (hunk 2 above)
Comment (pivnicek) [reply src/Keboola/DbExtractor/Extractor/PgSQL.php]: Yes, but inline comments break the `\copy` command

Comment (odinuv) [reply src/Keboola/DbExtractor/Extractor/PgSQL.php]: ok, then maybe put them at the begining and add a comment that that no comments can be made to comment that comment :D

seriously attnum > 0 is quite cryptic on itself

**Comment by pivnicek** [comment]
> seems legit, though i hate it when you change function in pref improvement

Yes, I know.  The reason I removed the defaults here is because going through csv there is no way to distinguish between `null` and `''`

File: src/Keboola/DbExtractor/Extractor/PgSQL.php (hunk 2 above)
Comment (pivnicek) [reply src/Keboola/DbExtractor/Extractor/PgSQL.php]: great idea, thanks!

**Pull request merged**